src/test/java/htsjdk/samtools/cram/encoding/ByteArrayLenCodecTest.java

```diff
@@ -0,0 +1,108 @@
+package htsjdk.samtools.cram.encoding;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.io.*;
+import htsjdk.samtools.cram.structure.EncodingID;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ByteArrayLenCodecTest extends HtsjdkTest {
```

Comment:
This test class is called `ByteArrayLenCodecTest`, but `ByteArrayLenCodec` is a private class inside of `ByteArrayLenEncoding`. That makes me think this should be called `ByteArrayLenEncodingTest. But I'm not sure if the tests should change, or the actual classes ? Is this is part of the weirdness you were talking about yesterday ?